internal/server/v1/event/event.go

```diff
@@ -177,16 +173,38 @@ func Collection(
 						continue
 					}
 				}
-				resp.EventCollection.Events = append(
-					resp.EventCollection.Events,
-					event,
-				)
-				provID := event.ProvenanceId
-				resp.EventCollection.ProvenanceInfo[provID] = data.ProvenanceInfo[provID]
+
+				eventID := event.GetDcid()
+				if _, ok := idToEvent[eventID]; !ok {
+					idToEvent[eventID] = event
```

Comment:
The event here is replaced by later import group? Since we rank the import groups already, so should keep the higher ranked ones, meaning continue here if already exists?

Reply:
NOTE that line 178 is "!ok", which means if not exist, we add, otherwise we don't add. So I think the code already achieves the logic you stated?

@shifucun

Reply:
I see